class.jetpack.php

```diff
@@ -7258,4 +7261,66 @@ public static function handle_post_authorization_actions( $activate_sso = false,
 
 		Jetpack::state( 'message', 'authorized' );
 	}
+
+	/**
+	 * Load Gutenberg editor blocks
+	 *
+	 * Blocks are disabled by default and enabled via filter:
+	 *   add_filter( 'jetpack_gutenberg', '__return_true', 10 );
+	 *
+	 * When enabled, blocks are loaded from CDN by default. To load locally instead:
+	 *   add_filter( 'jetpack_gutenberg_cdn', '__return_false', 10 );
+	 *
+	 * Note that when loaded locally, you need to build the files yourself:
+	 * - _inc/blocks/jetpack-editor.js
+	 * - _inc/blocks/jetpack-editor.css
+	 *
+	 * CDN cache is busted once a day or when Jetpack version changes. To customize it:
+	 *   add_filter( 'jetpack_gutenberg_cdn_cache_buster', function( $version ) { return time(); }, 10, 1 );
+	 */
+	public static function load_jetpack_gutenberg() {
+		if ( ! apply_filters( 'jetpack_gutenberg', false ) || ! Jetpack::is_plugin_active( 'gutenberg/gutenberg.php' ) ) {
+			return;
+		}
+
+		if ( apply_filters( 'jetpack_gutenberg_cdn', true ) ) {
+			$editor_script = 'https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks/jetpack-editor.js';
+			$editor_style = 'https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks/jetpack-editor.css';
+			$version = apply_filters( 'jetpack_gutenberg_cdn_cache_buster', sprintf( '%s-%s', gmdate( 'd-m-Y' ), JETPACK__VERSION ) );
+		} else {
+			$editor_script = plugins_url( '_inc/blocks/jetpack-editor.js', JETPACK__PLUGIN_FILE );
+			$editor_style = plugins_url( '_inc/blocks/jetpack-editor.css', JETPACK__PLUGIN_FILE );
+			$version = Jetpack::is_development_version() ? filemtime( JETPACK__PLUGIN_DIR . '_inc/blocks/jetpack-editor.js' ) : JETPACK__VERSION;
+		}
+
+		wp_register_script(
+			'jetpack-blocks-editor',
+			$editor_script,
+			array(
+				'wp-blocks',
+				'wp-components',
+				'wp-compose',
+				'wp-data',
+				'wp-editor',
+				'wp-element',
+				'wp-i18n',
+				'wp-plugins',
+			),
+			$version
+		);
+
+		wp_register_style(
+			'jetpack-blocks-editor',
+			$editor_style,
+			array(),
+			$version
+		);
+
+		if ( function_exists( 'register_block_type' ) ) {
```

Comment:
Can we have a method like `Jetpack::is_gutenberg_available` or such that wraps this check that probably will get used a lot ?